test/src/unit-cppapi-consolidation.cc

```diff
@@ -488,3 +489,51 @@ TEST_CASE(
   if (vfs.is_dir(array_name))
     vfs.remove_dir(array_name);
 }
+
+TEST_CASE(
+    "C++ API: Test consolidation that respects the current domain",
+    "[cppapi][consolidation][current-domain]") {
+  std::string array_name = "cppapi_consolidation_current_domain";
+  remove_array(array_name);
+
+  Context ctx;
+
+  Domain domain(ctx);
```

Comment:
You should probably mention, as a comment, that we're assuming the tile exent is bigger than the 2 & 3 here you're applying for current domain.

But actually, I think it would be even better to set the tile extent to 10, or 100, or 512, or whatever, explicitly when you create the test array.